pkg/sql/alter_table.go

```diff
@@ -405,7 +406,39 @@ func (n *alterTableNode) startExec(params runParams) error {
 
 			// Disable primary key changes on tables that are interleaved parents.
 			if len(n.tableDesc.PrimaryIndex.InterleavedBy) != 0 {
-				return errors.New("cannot change the primary key of an interleaved parent")
+				if len(n.tableDesc.PrimaryIndex.InterleavedBy) == 1 {
+					childTable, err := params.p.Tables().getTableVersionByID(
+						params.ctx, params.p.Txn(), n.tableDesc.PrimaryIndex.InterleavedBy[0].Table, tree.ObjectLookupFlags{})
```

Comment:
wonder if it's worth less duplicating code by having just the one error message, looks something like

```
statement error pq: cannot change primary key of table parent because table(s) [child, child2, child3] is interleaved into it
```

also minor nit: prefer one arg per line if there are multiple lines for a function call (but not sure if this is a crdb convention).

Reply:
I'm a fan, updated.